service/src/main/java/com/zebrunner/reporting/service/TestService.java

```diff
@@ -413,12 +414,21 @@ public boolean existsTestByIdAndTestRunId(long id, long testRunId) {
         return testMapper.getTestsByTestRunId(testRunId);
     }
 
-    public List<TestResult> getTestResultsByTestCaseId(Long testCaseId, Long limit) {
-        List<TestResult> testResults = testMapper.getTestResultsByTestCaseId(testCaseId, limit);
+    public List<TestResult> getLatestTestResultsByTestId(Long testId, Long limit) {
+        Test test = getNotNullTestById(testId);
+        List<TestResult> testResults = testMapper.getTestResultsByStartTimeAndTestCaseId(test.getTestCaseId(), test.getStartTime(), limit);
         testResults.forEach(result -> {
-            Duration elapsed = Duration.between(result.getStartTime(), result.getFinishTime());
-            result.setElapsed(elapsed.toMillis());
+            List<WorkItem> bugs = result.getWorkItems()
+                                        .stream()
+                                        .filter(workItem -> workItem.getType() == WorkItem.Type.BUG)
```

Comment:
change to equals (we shouldn't know how to work equals method inside)

Reply:
As I know enums are compared using == without any problems, some sources even mention this way as preferred